app/components/edit/EditNotes.tsx

```diff
@@ -12,139 +12,94 @@ export interface InternalNoteChanges {
   isRemoved?: boolean;
 }
 
+export type NotesObject = Record<number, InternalNoteChanges>;
+
 type Props = {
   notes: Note[] | undefined;
-  handleNotesChange: (n: State) => void;
-};
-
-export type State = {
-  notes: Record<number, InternalNoteChanges>;
-  existingNotes: InternalNote[];
-  uuid: number;
+  handleNotesChange: (notesObject: NotesObject) => void;
 };
 
-class EditNotes extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    this.state = {
-      notes: {},
-      existingNotes: props.notes
-        ? props.notes.map((note) => {
-            // HACK: This is bad because we are mutating the props to this
-            // component, which should never happen in a component. We should
-            // create a copy of the original Note before doing this, but before
-            // changing this code, we need to make sure that nothing else is relying on
-            // that mutation happening to the original Note props.
-            const newNote = note as InternalNote;
-            newNote.key = note.id;
-            return newNote;
-          })
-        : [],
-      uuid: -1,
+const EditNotes = ({ notes = [], handleNotesChange }: Props) => {
+  const internalNotes = notes.map((note) => {
+    const newNote: InternalNote = {
+      ...note,
+      key: note.id,
     };
 
-    this.renderNotes = this.renderNotes.bind(this);
-    this.handleNoteChange = this.handleNoteChange.bind(this);
-    this.addNote = this.addNote.bind(this);
-    this.removeNote = this.removeNote.bind(this);
-  }
-
-  handleNoteChange(key: number, note: InternalNoteChanges): void {
-    const { notes } = this.state;
-    const { handleNotesChange } = this.props;
-    notes[key] = note;
-    this.setState(
-      {
-        notes,
-      },
-      () => {
-        // HACK: This is really strange, to pass in the entirety of the State of
-        // this component to a change handler. We should really be more
-        // selectively deciding what state should be synchronized to the parent
-        // component.
-        handleNotesChange(this.state);
-      }
-    );
-  }
-
-  addNote() {
-    const { existingNotes, uuid } = this.state;
-    const newUUID = uuid - 1;
-    existingNotes.push({
-      key: newUUID,
+    return newNote;
+  });
+
+  const [notesList, setNotesList] = useState(internalNotes);
+  const [uuid, setUuid] = useState(-1);
+  const [notesObject, setNotesObject] = useState<NotesObject>({});
```

Comment:
Is `notesList` equivalent to the previous `existingNotes` state variable, and `notesObject` equivalent to the previous `notes` state variable? It'd be nice if we could unify the two state variables, since it seems a bit confusing to me to have both, but if that's too large of a refactor, I'm wondering if we can revert back to the original variable names? `existingNotes` at least implied that it was originally derived from the initial response from the API, but the current naming of `notesList` and `notesObject` makes it more difficult to understand which is which. Also, if it's not too much trouble, could we also rename `notesObject` to `newNotes`, to better differentiate it from `existingNotes`?

Reply:
Yep, `notesList` is equivalent to the previous `existingNotes`. And `notesObject` is equivalent to the previous `notes`. I _believe_ that our update to the API only reference `notes` aka `notesObject`, so trying to combine the two state vars might be a heavier lift than I feel comfortable doing as part of this PR.

But in any case, I've gone ahead and changed the names to your suggestion ones - makes good sense to me!